core/src/cluster_info_vote_listener.rs

```diff
@@ -430,6 +430,10 @@ impl ClusterInfoVoteListener {
                         verified_packets_sender,
                         &verified_vote_packets,
                     )?;
+                } else if let Some(bank_vote_sender_state) = bank_vote_sender_state_option {
+                    // This ensures we report the last slot's metrics
+                    bank_vote_sender_state.report_metrics();
+                    bank_vote_sender_state_option = None;
```

Comment:
nit: I think it might make sense to move all of this logic to `check_for_leader_bank_and_send_votes` and pass  the option `current_working_bank`